lib/iris/tests/unit/fileformats/grib/test_GribWrapper.py

```diff
@@ -88,90 +88,48 @@ def test_edition_1(self):
-        self.addCleanup(compute_patch.stop)
-        self.addCleanup(long_patch.stop)
-        self.addCleanup(string_patch.stop)
-        self.addCleanup(native_patch.stop)
-
-    def test_regular_sequential(self):
-        tell_tale = np.arange(1, 5) * _message_length
-        grib_fh = mock.Mock(tell=mock.Mock(side_effect=tell_tale))
-        grib_message = 'regular_ll'
-        for i, _ in enumerate(tell_tale):
-            gw = GribWrapper(grib_message, grib_fh)
-            self.assertTrue(is_lazy_data(gw._data))
-            proxy = as_concrete_data(gw._data)
-            self.assertIsInstance(proxy, GribDataProxy)
-            self.assertEqual(proxy.shape, (10, 20))
-            self.assertEqual(proxy.dtype, np.float)
-            self.assertIs(proxy.fill_value, np.nan)
-            self.assertEqual(proxy.path, grib_fh.name)
-            self.assertEqual(proxy.offset, _message_length * i)
-
-    def test_regular_mixed(self):
         tell_tale = np.arange(1, 5) * _message_length
-        expected = tell_tale - _message_length
-        grib_fh = mock.Mock(tell=mock.Mock(side_effect=tell_tale))
+        self.grib_fh = mock.Mock(tell=mock.Mock(side_effect=tell_tale))
+
+    def test_regular_proxy_args(self):
         grib_message = 'regular_ll'
-        for offset in expected:
-            gw = GribWrapper(grib_message, grib_fh)
-            self.assertTrue(is_lazy_data(gw._data))
-            proxy = as_concrete_data(gw._data)
-            self.assertIsInstance(proxy, GribDataProxy)
-            self.assertEqual(proxy.shape, (10, 20))
-            self.assertEqual(proxy.dtype, np.float)
-            self.assertIs(proxy.fill_value, np.nan)
-            self.assertEqual(proxy.path, grib_fh.name)
-            self.assertEqual(proxy.offset, offset)
-
-    def test_reduced_sequential(self):
-        tell_tale = np.arange(1, 5) * _message_length
-        grib_fh = mock.Mock(tell=mock.Mock(side_effect=tell_tale))
-        grib_message = 'reduced_gg'
-        for i, _ in enumerate(tell_tale):
-            gw = GribWrapper(grib_message, grib_fh)
-            self.assertTrue(is_lazy_data(gw._data))
-            proxy = as_concrete_data(gw._data)
-            self.assertIsInstance(proxy, GribDataProxy)
-            self.assertEqual(proxy.shape, (200,))
-            self.assertEqual(proxy.dtype, np.float)
-            self.assertIs(proxy.fill_value, np.nan)
-            self.assertEqual(proxy.path, grib_fh.name)
-            self.assertEqual(proxy.offset, _message_length * i)
-
-    def test_reduced_mixed(self):
-        tell_tale = np.arange(1, 5) * _message_length
-        expected = tell_tale - _message_length
-        grib_fh = mock.Mock(tell=mock.Mock(side_effect=tell_tale))
+        with mock.patch('iris.fileformats.grib.GribDataProxy') as mock_gdp:
+            gw = GribWrapper(grib_message, self.grib_fh)
+        mock_gdp.assert_called_once_with((10, 20), 'float64', np.nan,
+                                         self.grib_fh.name, 0)
```

Comment:
Yes, I think so.  I also think that (maybe) if I implement these extra things in the SetUp, then the tests will seem a little less circular.